manifests/init.pp

```diff
@@ -4,431 +4,560 @@
+#                                        type:boolean
 #
-# $server_enc_api::                   What version of enc script to deploy. Valid
-#                                     values are 'v2' for latest, and 'v1'
-#                                     for Foreman =< 1.2
+# $auth_allowed::                        An array of authenticated nodes allowed to
+#                                        access all catalog and node endpoints.
+#                                        default to ['$1']
+#                                        type:array
 #
-# $server_report_api::                What version of report processor to deploy.
-#                                     Valid values are 'v2' for latest, and 'v1'
-#                                     for Foreman =< 1.2
+# $server_default_manifest::             Toggle if default_manifest setting should
+#                                        be added to the [main] section
+#                                        type:boolean
 #
-# $server_request_timeout::           Timeout in node.rb script for fetching
-#                                     catalog from Foreman (in seconds).
-#                                     type:integer
+# $server_default_manifest_path::        A string setting the path to the default_manifest
+#                                        type:string
 #
-# $server_environment_timeout::       Timeout for cached compiled catalogs (10s, 5m, ...)
+# $server_default_manifest_content::     A string to set the content of the default_manifest
+#                                        If set to '' it will not manage the file
+#                                        type:string
 #
-# $server_ca_proxy::                  The actual server that handles puppet CA.
-#                                     Setting this to anything non-empty causes
-#                                     the apache vhost to set up a proxy for all
-#                                     certificates pointing to the value.
+# $server_ssl_dir_manage::               Toggle if ssl_dir should be added to the [master]
+#                                        configuration section. This is necessary to
+#                                        disable in case CA is delegated to a separate instance
+#                                        type:boolean
 #
-# $server_jvm_java_bin::              Set the default java to use.
+# $server_puppetserver_dir::             The path of the puppetserver config dir
+#                                        type:string
 #
-# $server_jvm_config::                Specify the puppetserver jvm configuration file.
+# $server_max_active_instances::         Max number of active jruby instances. Defaults to
+#                                        processor count
+#                                        type:integer
 #
-# $server_jvm_min_heap_size::         Specify the minimum jvm heap space.
+# $server_idle_timeout_milliseconds::    How long the server will wait for a response
```

Comment:
Not really opposed, I think in the last (way smaller) PRs it was ommitted for briefness. but anyay, explicit is probably also here better than implicit. @ares?